src/test/java/test/AdditionalFilesTest.java

```diff
@@ -34,8 +35,14 @@ void additional_files_config_map_creates_volumes(Product product) throws Excepti
                 "additionalFiles[0].mountPath", "/var/atlassian"
         ));
 
-        StatefulSet statefulSet = resources.getStatefulSet(product.getHelmReleaseName());
-        JsonNode configMap = getVolume(statefulSet, "configMap");
+        JsonNode configMap;
+        if (product.name().equals("bamboo_agent")) {
```

Comment:
I think this could be a different test - one for deployments (including only Bamboo agent) and the other for StatefulSets. The preparation part of setting up the `resources` could be a private function to remove duplication. I think it is a matter of personal preference, but I don't like `if` statements in the test functions.

Having said that I think this can stay as it is, it is conceptually testing the same thing so it is just a different tradeoff.

Reply:
Yeah, I was thinking about a dedicated test, though it's the same amount of duplicated code. Perhaps, once we switch to yaml based tests, things will become simpler.